numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
I think you have to `Py_INCREF(Py_None)` before passing it on. Since you then would have to `Py_DECREF` it as well, I suggest passing in `NULL` here, and doing the translation to `Py_None` above.